pkgs/stdenv/generic/setup.sh

```diff
@@ -1073,7 +1073,13 @@ _defaultUnpack() {
                 # stages. The XZ_OPT env var is only used by the full "XZ utils" implementation, which supports
                 # the --threads (-T) flag. This allows us to enable multithreaded decompression exclusively on
                 # that implementation, without the use of complex bash conditionals and checks.
-                XZ_OPT="--threads=$NIX_BUILD_CORES" xz -d < "$fn" | tar xf - --warning=no-timestamp
+                # Since tar does not control the decompression, we need to
+                # disregard the error code from the xz invocation. Otherwise,
+                # it can happen that tar exits earlier, causing xz to fail
+                # from a SIGPIPE. Since tar can detect whether the decompressed
+                # archive was complete, relying on tar is sufficient and more
+                # robust.
+                (XZ_OPT="--threads=$NIX_BUILD_CORES" xz -d < "$fn"; true) | tar xf - --warning=no-timestamp
```

Comment:
commit 61d0c5d931128e5471c6ed1e31cd0d6a2455fda0
Author: Eelco Dolstra <eelco.dolstra@logicblox.com>
Date:   Thu Jan 19 17:33:33 2012 +0000

    * Don't rely on tar knowing about .xz.

should we revisit and just use tar ?

Reply:
I was wondering why we don't just use `tar` directly. Seems weird not being able to rely on it in a reproducible environment. I think it makes sense to keep the change minimal at least for `staging-23.11` though.

Reply:
@zeuner I can't comment on the issue, since I've never experienced it. So please don't let my comment block this PR.

It would be good if @edolstra could comment. My guess is that historically there were BSD tar and GNU tar and they probably differed in behaviors. It would  be good if we could give it an other try ( maybe in different PR )

Reply:
That was 12 years old commit, so yes &ndash; perhaps it's perfectly fine now.